leader/leader.go

```diff
@@ -60,23 +59,22 @@ var log = logf.Log.WithName("leader")
 // attempts to become the leader.
 const maxBackoffInterval = time.Second * 16
 
-// Become ensures that the current pod is the leader within its namespace. If
-// run outside a cluster, it will skip leader election and return nil. It
+// Become ensures that the current pod is the leader within the given
+// namespace. If run outside a cluster, it will return an error. It
 // continuously tries to create a ConfigMap with the provided name and the
 // current pod set as the owner reference. Only one can exist at a time with
 // the same name, so the pod that successfully creates the ConfigMap is the
-// leader. Upon termination of that pod, the garbage collector will delete the
-// ConfigMap, enabling a different pod to become the leader.
-func Become(ctx context.Context, lockName string) error {
+// leader. Upon termination of that pod, the garbage collector will delete
+// the ConfigMap, enabling a different pod to become the leader.
+func Become(ctx context.Context, lockName string, ns string) error {
```

Comment:
WDYT of duplicating the [leaderelection.Options](https://github.com/kubernetes-sigs/controller-runtime/blob/229c3c357d9e6b07c3d6774010c35161a82b08f9/pkg/leaderelection/leader_election.go#L34-L47) struct from controller-runtime and using that as a parameter to `leader.Become()`

```suggestion
func Become(ctx context.Context, opts Options) error {
```